### PR TITLE
feat: metatx v1 endpoints

### PR DESCRIPTION
**Pull request by romainbutteaud**
# Description
- Rebase of @kimpers's work (https://github.com/0xProject/0x-api/pull/366) onto our v1-only branch (https://github.com/0xProject/0x-api/pull/367)
- Removing all meta_tx/v0 code 
- Matcha WIP integration available at https://github.com/0xProject/matcha-frontend/pull/1011

# Testing Instructions

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.

### CODE REVIEW

**Comment by kimpers** [comment]
deploy staging

**Comment by kimpers** [comment]
deploy staging

**Comment by kimpers** [comment]
deploy staging

**Comment by kimpers** [comment]
deploy staging

**Comment by kimpers** [comment]
deploy staging

**Comment by kimpers** [comment]
deploy staging

File: src/services/meta_transaction_service.ts [hunk 1]
```diff
@@ -424,55 +285,140 @@ export class MetaTransactionService {
...
+        .integerValue(BigNumber.ROUND_UP)
+        .times(ONE_GWEI);
+}
+
+function createExpirationTime(): BigNumber {
+    return new BigNumber(Date.now() + TEN_MINUTES_MS).div(ONE_SECOND_MS).integerValue(BigNumber.ROUND_CEIL);
+}
+
+function calculateProtocolFeeRequiredForOrders(gasPrice: BigNumber, orders: Array<SignedOrder | Order>): BigNumber {
+    const nativeOrderCount = orders.filter(o =>
```
Comment (dorothy-zbornak) [inline src/services/meta_transaction_service.ts:419]: Isn't this counting the bridge orders? Did I write this? lol :innocent:

File: src/types.ts [hunk 2]
```diff
@@ -531,6 +526,7 @@ export interface GetTransactionRequestParams {
     excludedSources?: ERC20BridgeSource[];
     includedSources?: ERC20BridgeSource[];
     includePriceComparisons: boolean;
+    affiliateFee: PercentageFee;
```
Comment (dorothy-zbornak) [inline src/types.ts:529]: :moneybag:

**Review by dorothy-zbornak — APPROVED** [review]
Looking dope. Looks good after some small requests!

File: src/services/meta_transaction_service.ts [hunk 3]
```diff
@@ -424,55 +285,140 @@ export class MetaTransactionService {
...
+            t => t.deploymentNonce.toString() === fillQuoteTransformerDeploymentNonce.toString(),
+        );
+
+        if (!fillQuoteTransformation) {
+            throw new Error('Could not find fill quote transformation in decoded calldata');
+        }
+
+        const decodedFillQuoteTransformerData = decodeFillQuoteTransformerData(fillQuoteTransformation.data);
+
+        return calculateProtocolFeeRequiredForOrders(mtx.minGasPrice, decodedFillQuoteTransformerData.orders);
```
[the remark this answers is not in the document]
Comment (kimpers) [reply src/services/meta_transaction_service.ts:384]: Yeah that makes sense, will fix 👍

**Comment by kimpers** [comment]
deploy staging

**Comment by kimpers** [comment]
deploy staging

**Comment by kimpers** [comment]
deploy staging

**Review by dekz — APPROVED** [review]
looks good. Appreciate the blood sweat and tears that went into this.

@johnrjj you ready for v1 transition soon?

**Comment by kimpers** [comment]
deploy staging

**Comment by kimpers** [comment]
deploy production

**Pull request merged**